### DISCUSSION TITLE
Support for Texas Instruments TIVA-C microcontrollers (TM4C123G, etc)

### DISCUSSION THREAD

**Discussion by 0xjairo** [zephyrproject-rtos/zephyr#47122] [Q&A]
Hello, what are people's opinions or thoughts on porting Zephyr to the Arm Cortex-M4F based microcontroller family from TI dubbed the Tiva C Series? These microcontrollers are very similar in specs to other vendor's Arm Cortex microcontrollers but you don't generally see them being used in the open source or maker community. Any thoughts as to why that might be?

Thanks

**Comment by Basilisvirus** [comment]
Id like to see it as well :)

**Comment by rbsexton** [comment]
Well, this is ancient.   

I use these devices at work.  I've shipped hundreds of thousands of them.    If Zephyr supported them I'd switch tomorrow.     The Tivas are really good little devices.  Easy to work with packages and really solid design and libraries.   I could see them sticking around for a long time in the industrial design space.

**Comment by rbsexton** [comment]
Here's an update:   I have an initial port of Zephyr running with UART, I2C, and Ethernet, along with pin map controls and Clock/PLL Init.     My target device is the TM4C129X parts.  

The SOC is a descendant of the original M3, the Stellaris, but that code is pretty old now and doesn't seem to match newer boards.    The Tiva has overlap with other devices, and I've used those as the basis for my port - Its got the UART and I2C IP from the SimpleLink devices, and the Ethernet is an older version of the Synopsis Designware IP thats already supported on other devices.       I combined that with TI's known good reference designs to capture workarounds for device errata.  

I'll do SPI/QSPI next and then ADCs, because I need those.  I'm up for doing more, but I might not have hardware to test with real devices

**Reply by lcarricart** [reply]
Hi! I do have a TM4C129EXL, and I'm interested in making my first Zephyr contribution, connected to board bring-up. Let me know if I can support you with one of your desired features, or if something else changed along the way!
